source/script.cpp

```diff
@@ -5953,21 +6004,43 @@ ResultType Script::DefineFunc(LPTSTR aBuf, bool aStatic, bool aIsInExpression)
 					if (*param_end == in_quote && param_end[-1] != '`')
 					{
 						++param_end;
+						// Check if this string is the first value in a concatenation
+						// eg, f(x: = 'y' z), handle as PARAM_DEFAULT_EXPR below.
+						auto next_non_white = *omit_leading_whitespace(param_end);
+						if (next_non_white == ','
+							|| next_non_white == ')')
+							is_default_str = true;
+						// else it is concatenation,
 						break;  // The previous char is the ending quote.
 					}
 					//else an escaped '"' or some character other than '\0' or '"'.
 					*target++ = *param_end++;
 				}
-				*target = '\0'; // Terminate it in the buffer.
-				// The above has also set param_end for use near the bottom of the loop.
-				ConvertEscapeSequences(buf, NULL); // Raw escape sequences like `n haven't been converted yet, so do it now.
-				this_param.default_type = PARAM_DEFAULT_STR;
-				this_param.default_str = *buf ? SimpleHeap::Alloc(buf, target-buf) : _T("");
+				if (is_default_str)
+				{ 
+					*target = '\0'; // Terminate it in the buffer.
+					// The above has also set param_end for use near the bottom of the loop.
+					ConvertEscapeSequences(buf, NULL); // Raw escape sequences like `n haven't been converted yet, so do it now.
+					this_param.default_type = PARAM_DEFAULT_STR;
+					this_param.default_str = *buf ? SimpleHeap::Alloc(buf, target-buf) : _T("");
+				}
 			}
-			else // A default value other than a quoted/literal string.
-			{
-				if (!(param_end = StrChrAny(param_start, _T(", \t=)")))) // Somewhat debatable but stricter seems better.
-					return ScriptError(ERR_MISSING_COMMA, aBuf); // Reporting aBuf vs. param_start seems more informative since Vicinity isn't shown.
+			if (!is_default_str) // A default value or expression other than a quoted/literal string.
+			{	
+				auto len_expr = FindExprDelim(param_start, ',');
+				if (param_start[len_expr] == '\0')
+				{
+					len_expr = FindExprDelim(param_start, ')');
```

Comment:
Note that a single call to `FindExprDelim(param_start, 0)` would find the first delimiter that can't be part of the expression. It is used by ParseFatArrow, for instance, to find the end of the fat arrow function within a larger expression.